### PR TITLE
Make colorpicker radiobuttons in a fieldset

### PR DESCRIPTION
**Pull request by RoyEJohnson**
For: https://github.com/openstax/unified/issues/2759
For: https://openstax.atlassian.net/browse/DISCO-96

### CODE REVIEW

File: src/app/content/highlights/components/ColorPicker.tsx [hunk 1]
```diff
@@ -62,33 +63,88 @@ const ColorButton = styled(({className, size, style, ...props}: ColorButtonProps
...
+      ref={ref as unknown as LegacyRef<HTMLElement>}
+      onKeyDown={handleKeyNavigation}
+      onFocus={initialFocus}
+    >
+      {highlightStyles.map((style) => <ColorButton key={style.label}
+        name={style.label}
+        checked={props.multiple ? props.selected.includes(style.label) : props.color === style.label}
+        style={style}
+        size={props.size}
+        tabindex={-1}
```
Comment (jivey) [inline src/app/content/highlights/components/ColorPicker.tsx:130]: React wants this to be `tabIndex` which I never remember until I randomly notice it yelling at me in the console

File: src/app/content/highlights/components/ColorPicker.tsx [hunk 2]
```diff
@@ -62,33 +63,88 @@ const ColorButton = styled(({className, size, style, ...props}: ColorButtonProps
...
-          : props.onChange(style.label)}
-    />)}
-  </div>;
+  return (
+    <div
+      className={className}
+      tabIndex={0}
+      role='radiogroup'
+      aria-label='colors'
+      ref={ref as unknown as LegacyRef<HTMLElement>}
```
Comment (jivey) [inline src/app/content/highlights/components/ColorPicker.tsx:121]: Wouldn't initializing the ref to `null` make this unnecessary?

**Review by jivey — CHANGES_REQUESTED** [review]

**Comment by jivey** [comment]
Not a change request, just wondering to expand my React knowledge... do you think using callback refs would be a more "Reacty" way of approaching this? I was peeking at https://react.dev/learn/manipulating-the-dom-with-refs#how-to-manage-a-list-of-refs-using-a-ref-callback but I dunno if it has render tradeoffs and might be too complicated for this.

File: src/app/content/highlights/components/ColorPicker.tsx (hunk 2 above)
Comment (RoyEJohnson) [reply src/app/content/highlights/components/ColorPicker.tsx:121]: That actually does work. I thought the differing types were the problem. Thanks.

**Comment by RoyEJohnson** [comment]
I think callback refs could be used to avoid making separate components where the only reason for the separate component was to allow it to have a ref. Since we already have a ColorButton, I don't see it offering an advantage here. My inclination is to make separate components, anyway, because it's more straightforward. There would probably be an efficiency advantage to callback refs if the list of sub-units was very long.

**Review by jivey — APPROVED** [review]
Looks good but you've got some lint and snapshot failures

**Pull request merged**